pkg/integration/components/env.go

```diff
@@ -0,0 +1,40 @@
+package components
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	GIT_CONFIG_GLOBAL_ENV_VAR = "GIT_CONFIG_GLOBAL"
+	LAZYGIT_ROOT_DIR          = "LAZYGIT_ROOT_DIR"
+	PATH                      = "PATH"
+	PWD                       = "PWD"
+	SANDBOX_ENV_VAR           = "SANDBOX"
+	TERM                      = "TERM"
+	TEST_NAME_ENV_VAR         = "TEST_NAME"
+	WAIT_FOR_DEBUGGER_ENV_VAR = "WAIT_FOR_DEBUGGER"
+)
+
+// These environment variables must be set for lazygit to run.
+// As a result, each variable in this list will be passed through
+// to integration tests.
+var environmentWhitelist = [...]string{
+	GIT_CONFIG_GLOBAL_ENV_VAR,
```

Comment:
Sorry, I think I neglected to mention in the PR discussion (but maybe I remembered in a commit message??) that I ran into an issue where (iirc) GIT_GLOBAL_CONFIG was getting set as an env var, but later cleared by the new test environment code. I decided to simply whitelist all of these env vars as policy, so that the timing and method of their setting would never break a test.

I think that's preferable to needing to pick out _which_ test-required environment variables need to be passed on based on the current implementation, but that's certainly up for debate. One way or another, only whitelisting PATH and TERM will cause some tests to break as the code stands.